src/Torch/Layer/NonLinear.hs

```diff
@@ -20,9 +20,13 @@ decodeAct :: ActName -> Tensor -> Tensor
 decodeAct actname = case actname of
                    Id  -> id
                    Sigmoid -> sigmoid
+                   Softmax -> softmax
                    Tanh -> tanh
                    Relu -> relu
                    Elu -> elu'
                    Selu -> selu
   -- | HACK : Torch.Functional.tanhとexpが `Segmentation fault`になるため
-  where tanh x = 2 * (sigmoid $ 2 * x) - 1
\ No newline at end of file
+  where tanh x = 2 * (sigmoid $ 2 * x) - 1
+        softmax x = Torch.Functional.div expInput expSum
```

Comment:
nit: You can use infix notation and remove Torch.Functional if you hide div from Prelude like tanh
```suggestion
        softmax x = expInput `div` expSum
```